### PR TITLE
[nv16] propagate gas outputs from fvm + integrate safer ffi

### PR DESCRIPTION
**Pull request by vyzo**
See https://github.com/filecoin-project/ref-fvm/issues/523
Depends on https://github.com/filecoin-project/filecoin-ffi/pull/275

Integrates https://github.com/filecoin-project/filecoin-ffi/pull/247.

### CODE REVIEW

File: extern/sector-storage/ffiwrapper/sealer_test.go [hunk 1]
```diff
@@ -869,89 +868,6 @@ func TestAddPiece512MPadded(t *testing.T) {
 	require.Equal(t, "baga6ea4seaqonenxyku4o7hr5xkzbqsceipf6xgli3on54beqbk6k246sbooobq", c.PieceCID.String())
 }
 
-func setupLogger(t *testing.T) *bytes.Buffer {
```
Comment (raulk) [inline extern/sector-storage/ffiwrapper/sealer_test.go:872]: Could you explain why this needs to go entirely? I understand that `generated` no longer exists, but has all this functionality vanished from the library? Or is it just behind different APIs? If the latter, we should adapt the tests, not drop the coverage.

**Review by raulk — COMMENTED** [review]
Tests that are failing are already failing.

File: extern/sector-storage/ffiwrapper/sealer_test.go (hunk 1 above)
Comment (vyzo) [reply extern/sector-storage/ffiwrapper/sealer_test.go:872]: its gone completely afaict.

Comment (raulk) [reply extern/sector-storage/ffiwrapper/sealer_test.go:872]: It isn't. AFAIK no functionality has been removed, only APIs have changed. Here's where the log is now:

filecoin-project/filecoin-ffi@bb281b5/cgo/util.go#L11

Comment (vyzo) [reply extern/sector-storage/ffiwrapper/sealer_test.go:872]: ok feel free to reinstate, it is really ugly however.

Comment (raulk) [reply extern/sector-storage/ffiwrapper/sealer_test.go]: Agreed that the form of the test is terrible, but it is providing certain coverage that we can't drop here.

**Pull request merged**